nix/output.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, name ? "default-output", ... }:
+with lib;
+{
+  options = {
+    name = mkOption {
+      default = "${name}";
+      type = types.str;
+      description = "Name of the output.";
+    };
+    script = mkOption {
+      default = null;
+      type = types.nullOr (types.str);
+      #type = types.nullOr (types.either types.str types.path);
+      description = ''
+        Text of a script which will produce a JSON value.
+        Warning: This uses shell features and is potentially dangerous.
+        Environment variables: 
+            $out is a temp directory available for use.
+        '';
```

Comment:
nitpick: Indentation, also please note that descriptions are written in Docbook, so maybe use [`<warning/>`](https://tdg.docbook.org/tdg/5.0/warning.html) and probably [`<envar/>`](https://tdg.docbook.org/tdg/5.0/envar.html) here.